src/sdk-reference/cpp/1/document/count/index.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: sdk.html.hbs
+algolia: true
+title: count
+description:
+order: 200
+---
+
+# count
+
+Given some filters, gets the number of matching documents from Kuzzle.
+
+Kuzzle uses the [ElasticSearch Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/5.x/query-dsl.html) syntax.
+
+## Arguments
+
+```cpp
+int count(
+    const std::string& index, 
+    const std::string& collection, 
+    const std::string& body, 
```

Comment:
Then we have a problem with the API. The body is mandatory both in kuzzle and in the go sdk:
https://github.com/kuzzleio/sdk-go/blob/master/document/count.go#L38
https://github.com/kuzzleio/kuzzle/blob/master/lib/api/controllers/documentController.js#L140

Reply:
I just checked the code in Kuzzle... The problem may be a bit deeper than what we both thought.

1. Kuzzle does check if a non-null body is provided, thus your documentation is correct, and the new API one isn't
2. But if an empty body is provided, then `document:count` does return the total number of documents in a collection

To me, there is a consistency problem : it seems weird to force the presence of a body where an empty body has a defined behavior. 
I still think the new API documentation is correct, and that both the SDK documentation and Kuzzle's code should be fixed, to make the body optional.

Reply:
That being said, the current Kuzzle state requires a body. So the best course of action may be to, first, change the new API documentation to make it match the current state.

And then to write a new issue, to change this behavior on a global scale (Kuzzle core + all SDK)
Then again, since SDK are currently in the work, it may be best to change SDK right now.

Reply:
I agree the behaviour of this method could/should be refactor to make the body optional.

I do not agree on your proposal regarding the sdk documentation though. The sdk update should follow kuzzle + api changes and not the opposite way round. 
Considering the priorities, I doubt refactor this method is urgent enough to be done before the documentation is released (and should not be anyway).

Reply:
Sounds good to me. I'm writing an issue and I'll propose a new PR on the API doc to fix it.